### ISSUE TITLE
Explanations for "Strings can be tricky sometimes!" are incomplete

### ISSUE THREAD

**Issue by jseakle** [satwikkansal/wtfpython#67]
The main issue is that they do not address the third snippet at all, "a" * 20 vs "a" * 21.

I'm also very curious to know /why/ it's the case that

> Strings that are not composed of ASCII letters, digits or underscores, are not interned. This explains why 'wtf!' was not interned due to !.

If it were just non-ASCII, that would make some sense, but what problems could interning a string with an exclamation point cause?

**Comment by sohaibfarooqi** [comment]
You can read [This SO](https://stackoverflow.com/a/24245514/5741172) post for details. It covers everything you need to know.

**Comment by jseakle** [comment]
The existence of an SO post doesn't make the explanation section isn't incomplete. The main issue is that it does not address the third snippet at all, "a" * 20 vs "a" * 21.

**Comment by sohaibfarooqi** [comment]
@jseakle I completely agree with you. Even I noticed this too when I was reading the explanation portion. Then I searched on SO and found the post that I posted in my earlier comment. 

Someone really have to improve the explanation for this portion.

**Comment by sheljohn** [comment]
+1, I was looking for the same explanation :) Would @satwikkansal be able to add a few words about this? Would a PR make it easier?

**Comment by satwikkansal** [comment]
Alright guys, really sorry for the ignorance. I've fixed this up in  a0100a8  Let me know if things are still unclear.

**Comment by sheljohn** [comment]
@satwikkansal Just submitted a PR about this :) Should I delete it? I think it would be good to mention that `is` should not be used for value comparison.

**Comment by satwikkansal** [comment]
> If it were just non-ASCII, that would make some sense, but what problems could interning a string with an exclamation point cause?

A look at the source code shows that this is defined [here](https://github.com/python/cpython/blob/078f1814f1a4413a2a0fdb8cf4490ee0fc98ef34/Objects/codeobject.c#L13) used [here](https://github.com/python/cpython/blob/078f1814f1a4413a2a0fdb8cf4490ee0fc98ef34/Objects/codeobject.c#L59) to decide if a string should be interned or not.

**Comment by satwikkansal** [comment]
@sheljohn oops, I didn't knew that, I'm having a look at your PR (y)

**Comment by satwikkansal** [comment]
>  Should I delete it? I think it would be good to mention that is should not be used for value comparison.

@sheljohn left a few comments at the PR. I guess maybe we should only go with the typo (fullstop) change for the time being.
